core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -619,6 +618,29 @@ public PlanOptimizers(
                                 new PruneTableScanColumns(metadata),
                                 new PushPredicateIntoTableScan(metadata, typeOperators, typeAnalyzer))));
 
+        // Try to pushdown aggregations before MultipleDistinctAggregationToMarkDistinct.
+        // This needs to be done after redirections.
+        builder.add(
```

Comment:
We should try to place this rules in existing optimizers if possible. Otherwise, we’re slowly moving back to the phased while-plan transformation strategy that we’ve being trying to replace.